pyprof/parse/nsight.py

```diff
@@ -224,7 +248,7 @@ def filterTrace(mlist):
             if m.find("CheckpointFunctionBackward") >= 0:
                 continue
 
-            if ("_backward, seq =" in m) or ("Backward, seq =" in m) or ("Backward0, seq =" in m):
+            if re.search(r"_backward, seq =|Backward[0-9]*, seq=", m):
```

Comment:
This is very delicate code. A space is missing between `seq=` which will break a lot of code (probably not because the code is resilient) but will result in incorrect correlation. The current QA of lenet is probably not effective. Can we move any refactoring to a different PR, so that its easy to roll back if errors arise.

Reply:
We should use a regex which matches the code i.e. `Backward` and `Backward0` as opposed to generalizing it to match 0 through 9. Reversing from a relaxed pattern to a stricter pattern will become impossible in the future for fear of breaking code.

Reply:
ok - so it sounds like there is not sufficient testing in place to support refactoring code?

I usually refactor as a I go.  I can put this back, but the chances of me later coming back to refactor it are pretty low.

Reply:
Yes, please put it back or move it to a separate PR. As you can imagine a lot of code (especially during backprop) is based on heuristics and it tries to decipher undocumented names/patterns used by the PyTorch backend. That's the part which is delicate. The purpose of hundreds of assertions is to catch errors when any of those assumptions no longer hold true. Adding a test for every heuristic can be done but requires a lot of effort.